agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -248,6 +271,140 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
         .addShutdownHook(new Thread(() -> flushAll(telemetryClient).join(10, TimeUnit.SECONDS)));
   }
 
+  private static TelemetryItemExporter initStatsbeatTelemetryItemExporter(
+      StatsbeatModule statsbeatModule, File tempDir, int diskPersistenceMaxSizeMb) {
+    HttpPipeline httpPipeline = LazyHttpClient.newHttpPipeLine(null);
+    TelemetryPipeline telemetryPipeline = new TelemetryPipeline(httpPipeline);
+
+    TelemetryPipelineListener telemetryPipelineListener;
+    if (tempDir == null) {
+      telemetryPipelineListener = new StatsbeatTelemetryPipelineListener(statsbeatModule::shutdown);
+    } else {
+      LocalStorageTelemetryPipelineListener localStorageTelemetryPipelineListener =
+          new LocalStorageTelemetryPipelineListener(
+              diskPersistenceMaxSizeMb,
+              TempDirs.getSubDir(tempDir, STATSBEAT_FOLDER_NAME),
+              telemetryPipeline,
+              LocalStorageStats.noop(),
+              true);
+      telemetryPipelineListener =
+          TelemetryPipelineListener.composite(
+              new StatsbeatTelemetryPipelineListener(
+                  () -> {
+                    statsbeatModule.shutdown();
```

Comment:
my guess is `shutdown` is being called somehow, which is terminating the ExecutorService